### PR TITLE
Use birth time instead of modify time for generation "built on" date

### PR DESCRIPTION
**Pull request by adtya**
PR towards #136

### CODE REVIEW

File: rust/tool/src/generation.rs [hunk 1]
```diff
@@ -96,7 +95,7 @@ impl fmt::Display for Generation {
 }
 
 fn read_build_time(path: &Path) -> Result<Date> {
-    let build_time = time::OffsetDateTime::from_unix_timestamp(fs::metadata(path)?.mtime())?.date();
+    let build_time = time::OffsetDateTime::from(fs::metadata(path)?.created().unwrap()).date();
```
Comment (blitz) [inline rust/tool/src/generation.rs:98]: Is it necessary to unwrap here instead of propagating the error?

Comment (adtya) [reply rust/tool/src/generation.rs:98]: Thanks for the feedback.
I'm a rust newbie and didn't know the difference between unwrap and propagating at the time.

you're right. there's no need to unwrap there. I'll update the PR.

Comment (blitz) [reply rust/tool/src/generation.rs]: Kudos that you gave it a shot! I would wait a bit for @nikstur to look at this.

File: rust/tool/src/generation.rs [hunk 2]
```diff
@@ -96,7 +95,7 @@ impl fmt::Display for Generation {
 }
 
 fn read_build_time(path: &Path) -> Result<Date> {
-    let build_time = time::OffsetDateTime::from_unix_timestamp(fs::metadata(path)?.mtime())?.date();
+    let build_time = time::OffsetDateTime::from(fs::metadata(path)?.created()?).date();
```
Comment (RaitoBezarius) [inline rust/tool/src/generation.rs:98]: I think you don't need all this, try:
```suggestion
    let build_time: time::OffsetDateTime = fs::metadata(path)?.created()?.into();
```
I would assume that a proper `Into<T>` impl is in the context.

Comment (RaitoBezarius) [reply rust/tool/src/generation.rs:98]: Probably: `Ok(fs::metadata(path)?.created()?.into())` would work too.

Comment (adtya) [reply rust/tool/src/generation.rs:98]: Thanks for the feedback.

Sure, I'll try this.
I was trying to get this to work with minimal changes, so I didn't want to modify anything outside getting the date to work.

**Review by blitz — APPROVED** [review]
Looks good to me!

**Comment by blitz** [comment]
Tests passed.

**Pull request merged**